### PR TITLE
[DI-25391] - Add firewall filters

### PR DESCRIPTION
**Pull request by ankita-akamai**
## Description 📝

Handle new service-type -`firewall` in dashboards.

## Changes  🔄

- Add filers - `firewalls`, `interface id`, and `interface type` for `firewall` servicetype.
- Add unit tests and update mocks.

## Target release date 🗓️
15 July 2025

## Preview 📷

Figma: https://docs.google.com/spreadsheets/d/1oPSiIJc-tvw7k88s4kd5b9Ns169g3VHgBbt8zDL-n0U/edit?pli=1&gid=1312373850#gid=1312373850

_*Figma might update, queries waiting for reply_

| Before  | After   |
| ------- | ------- |
| Firewalls didn't exist in dahsboards | image |
| Firewallss didn't exist in dahsboards | image |

To be contextual view [not integrated officially yet]:
image

filters=> (focus on interface id and type filter)
image

### Verification steps

- Navigate to metrics tab under monitor `/metrics`.
- Select dashboard - firewall dashboard.
- Play around with the filters and validate above screenshots and changes mentioned.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

*Check all that apply*

- [ ] Use React components instead of HTML Tags
- [ ] Proper naming conventions like cameCase for variables & Function & snake_case for constants
- [ ] Use appropriate types & avoid using "any"
- [ ] No type casting & non-null assertions
- [ ] Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] Providing/Improving test coverage
- [ ] Use sx props to pass styles instead of style prop
- [ ] Add JSDoc comments for interface properties & functions
- [ ] Use strict equality (===) instead of double equal (==)
- [ ] Use of named arguments (interfaces) if function argument list exceeds size 2
- [ ] Destructure the props
- [ ] Keep component size small & move big computing functions to separate utility
- [ ] 📱 Providing mobile support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

---

### CODE REVIEW

**Comment by ankita-akamai** [comment]
This needs to be rebased after dev-sync merge.

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      placeholder,
+      handleTextFilterChange,
+      savePreferences,
+      defaultValue,
+      disabled,
+      filterKey,
+    } = props;
+
+    const [value, setValue] = React.useState<string>(
+      (defaultValue as string) || ''
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: why typecasting?

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  validationFunction,
+} from '../Utils/constants';
+
+import type { Dashboard, FilterValue } from '@linode/api-v4';
+
+export interface CloudPulseTextFilterProps {
+  /**
+   * The dashboard object
+   */
+  dashboard: Dashboard;
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: what is the use of this prop?

**Comment by ankita-akamai** [comment]
Working on test failures

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx (hunk 2 above)
Comment (ankita-akamai) [reply packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: used earlier for some purpose then it got removed, will remove the prop as well, thanks

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      <TextField
+        autoComplete="off"
+        disabled={disabled}
+        errorText={errorText}
+        helperText={!errorText ? HELPER_TEXT[filterKey] : undefined}
+        label={label}
+        noMarginTop
+        onBlur={handleBlur}
+        onChange={handleInputChange}
+        optional
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: @ankita-akamai , i really don't understand this, i feel we should drive the optional or required from FilterConfig right, but here i don't see those.

Also how did the UT's didn't capture this? Are we missing this cases in UT?

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.test.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+const defaultPropsInterfaceId: CloudPulseTextFilterProps = {
+  dashboard: dashboardFactory.build(),
+  handleTextFilterChange: mockHandleTextFilterChange,
+  label: 'Interface IDs',
+  savePreferences: false,
+  filterKey: 'interface_id',
+  placeholder: INTERFACE_IDS_PLACEHOLDER_TEXT,
+};
+
+describe('CloudPulseTextFilter for port', () => {
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.test.tsx:42]: @ankita-akamai , lets improve the UT's to cover each and every edge cases as well

**Review by venkymano-akamai — CHANGES_REQUESTED** [review]
Please see comments, will review sometime later

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx (hunk 3 above)
Comment (ankita-akamai) [reply packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: UTs weren't missing the check because 'optional' wasn't derived from config, so directly checked for optional, will derive from config and update uts

File: packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx (hunk 1 above)
Comment (ankita-akamai) [reply packages/manager/src/features/CloudPulse/shared/CloudPulseTextFilter.tsx]: already approved code but i will remove

**Review by venkymano-akamai — APPROVED** [review]
Nice!!

**Review by nikhagra-akamai — APPROVED** [review]
